### PR TITLE
Add CodeQL workflow

### PR DESCRIPTION
**Pull request by n1c00o**
Add CodeQL to automatically scan for vulnerabilities in Go code.
I am gonna work on some other workflows and maybe replace this one but we still need security audits for now.

### CODE REVIEW

**Comment by n1c00o** [comment]
'we need to be careful about how the workflow is integrated with our build tools)

**Review by icecodder — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (MatthieuCoder) [inline .github/workflows/codeql-analysis.yml:57]: can you replace this with a bazel command ? like we do on the main branch

**Comment by MatthieuCoder** [comment]
CI build can be ignored

**Comment by MatthieuCoder** [comment]
LGTM

**Comment by MatthieuCoder** [comment]
Cn you merge it into `new-gateway` ?

**Comment by MatthieuCoder** [comment]
Merged in #4

**Pull request closed without merging**